javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,240 @@
+
+  def removeLast(): E =
+    if (inner.isEmpty) throw new NoSuchElementException()
+    else inner.remove(inner.size - 1).inner
+
+  def addFirst(e: E): Unit =
+    Box(e) +=: inner
+
+  def addLast(e: E): Unit =
+    inner += Box(e)
+
+  override def contains(o: Any): Boolean = {
+    val boxed = Box(o.asInstanceOf[E])
+    inner.contains(boxed)
+  }
+
+  override def size(): Int =
+    inner.size
+
+  override def add(e: E): Boolean = {
+    addLast(e)
+    true
+  }
+
+  override def remove(o: Any): Boolean = {
+    val sizeBefore = inner.size
+    inner -= Box(o.asInstanceOf[E])
+    sizeBefore != inner.size
+  }
+
+  override def addAll(c: Collection[_ <: E]): Boolean = {
+    val iter = c.iterator
+    var changed = iter.hasNext()
+    while (iter.hasNext())
+      addLast(iter.next())
+
+    changed
+  }
+
+  override def clear(): Unit =
+    inner.clear()
+
+  override def get(index: Int): E =
+    if (index < 0 || index >= size()) throw new IndexOutOfBoundsException(index.toString)
+    else inner(index).inner
+
+  override def set(index: Int, element: E): E = {
+    add(index, element)
+    remove(index+1)
+  }
+
+  override def add(index: Int, element: E): Unit =
+    if (index < 0 || index > size()) throw new IndexOutOfBoundsException(index.toString)
+    else inner.insert(index, Box(element))
+
+  override def remove(index: Int): E =
+    if (index < 0 || index >= size()) throw new IndexOutOfBoundsException(index.toString)
+    else inner.remove(index).inner
+
+  override def indexOf(o: Any): Int =
+    inner.indexOf(Box(o.asInstanceOf[E]))
+
+  override def lastIndexOf(o: Any): Int =
+    inner.lastIndexOf(Box(o.asInstanceOf[E]))
+
+  def peek(): E =
+    peekFirst()
+
+  def element(): E =
+    getFirst()
+
+  def poll(): E =
+    pollFirst()
+
+  def remove(): E =
+    remove(0)
+
+  def offer(e: E): Boolean =
+    offerLast(e)
+
+  def offerFirst(e: E): Boolean = {
+    addFirst(e)
+    true
+  }
+
+  def offerLast(e: E): Boolean = {
+    addLast(e)
+    true
+  }
+
+  def peekFirst(): E =
+    inner.headOption.fold(null.asInstanceOf[E])(_.inner)
+
+  def peekLast(): E =
+    inner.lastOption.fold(null.asInstanceOf[E])(_.inner)
+
+  def pollFirst(): E = {
+    val ret = peekFirst()
+    if (ret == null) null.asInstanceOf[E] else {
+      inner.trimStart(1)
+      ret
+    }
+  }
+
+  def pollLast(): E = {
+    val ret = peekLast()
+    if (ret == null) null.asInstanceOf[E] else {
+      inner.trimEnd(1)
+      ret
+    }
+  }
+
+  def push(e: E): Unit =
+    addFirst(e)
+
+  def pop(): E =
+    removeFirst()
+
+  def removeFirstOccurrence(o: Any): Boolean = {
+    val index = indexOf(o)
+    if (index < 0 || index >= size()) false else {
+      remove(index)
+      true
+    }
+  }
+
+  def removeLastOccurrence(o: Any): Boolean = {
+    val index = lastIndexOf(o)
+    if (index < 0 || index >= size()) false else {
+      remove(index)
+      true
+    }
+  }
+
+  override def listIterator(_index: Int): ListIterator[E] = {
```

Comment:
Note: the implementation that you did here is actually going to do the same as the one that your overloading and therefore it is useless to override. Also, as sjrd said, the complexities should be O(1) and therefore the only way to do this in a linked list is using it's the reference to the current node. To have access to the node you will have to create your own classes to be able to access the internal structure and at the same time hiding it.